src/mito2/src/row_converter.rs

```diff
@@ -0,0 +1,409 @@
+            Int64, i64,
+            UInt64, u64,
+            Float32, f32,
+            Float64, f64,
+            String, string,
+            Date, date,
+            DateTime, datetime,
+            Time, time,
+            Interval, interval
+        );
+
+        Ok(())
+    }
+
+    fn deserialize<B: Buf>(&self, deserializer: &mut Deserializer<B>) -> error::Result<Value> {
+        use common_time::DateTime;
+        macro_rules! deserialize_and_build_value {
+            (
+                $self: ident;
+                $serializer: ident;
+                $(
+                    $ty: ident, $f: ident
+                ),*
+            ) => {
+
+                match &$self.data_type {
+                    $(
+                        ConcreteDataType::$ty(_) => {
+                            Ok(Value::from(Option::<$f>::deserialize(deserializer).context(error::DeserializeFieldSnafu)?))
+                        }
+                    )*
+                    ConcreteDataType::Binary(_) => Ok(Value::from(
+                        Option::<Vec<u8>>::deserialize(deserializer)
+                            .context(error::DeserializeFieldSnafu)?
+                            .map(Bytes::from),
+                    )),
+                    ConcreteDataType::Timestamp(_) => {
+                        let timestamp = Option::<MemComparableTimestamp>::deserialize(deserializer)
+                            .context(error::DeserializeFieldSnafu)?
+                            .map(Timestamp::try_from)
+                            .transpose()?;
+                        Ok(Value::from(timestamp))
+                    }
+                    ConcreteDataType::List(l) => NotSupportedFieldSnafu {
+                        data_type: ConcreteDataType::List(l.clone()),
+                    }
+                    .fail(),
+                    ConcreteDataType::Dictionary(d) => NotSupportedFieldSnafu {
+                        data_type: ConcreteDataType::Dictionary(d.clone()),
+                    }
+                    .fail(),
+                    ConcreteDataType::Null(n) => NotSupportedFieldSnafu {
+                        data_type: ConcreteDataType::Null(n.clone()),
+                    }
+                    .fail(),
+                }
+            };
+        }
+        deserialize_and_build_value!(self; deserializer;
+            Boolean, bool,
+            Int8, i8,
+            Int16, i16,
+            Int32, i32,
+            Int64, i64,
+            UInt8, u8,
+            UInt16, u16,
+            UInt32, u32,
+            UInt64, u64,
+            Float32, f32,
+            Float64, f64,
+            String, String,
+            Date, Date,
+            Time, Time,
+            DateTime, DateTime,
+            Interval, Interval
+        )
+    }
+}
+
+/// A memory-comparable representation for [Timestamp] that is used for
+/// serialization/deserialization.
+#[derive(Debug, Serialize, Deserialize)]
+struct MemComparableTimestamp {
```

Comment:
Could we just encode timestamps as `i64`? I remember that we just i64 array for timestamps.

Reply:
> Could we just encode timestamps as `i64`? I remember that we just i64 array for timestamps.

That depends on whether we need to preserve of the order of timestamps with different time unit.

Should we consider encoding and comparing timestamps with different units?